providers/collection_package.rb

```diff
@@ -92,3 +119,129 @@ def create_collection_package
 
   Opennms::Helpers.write_xml_file(doc, "#{node['opennms']['conf']['home']}/etc/collectd-configuration.xml")
 end
+
+def update_collection_package
+	Chef::Log.debug "Updating collectd-configuration.xml definition : '#{new_resource.name}'"
+	file = ::File.new("#{node['opennms']['conf']['home']}/etc/collectd-configuration.xml")
+	contents = file.read
+	doc = REXML::Document.new(contents, respect_whitespace: :all)
+	doc.context[:attribute_quote] = :quote
+	file.close
+	
+	package_el = matching_package_name(doc, new_resource)
+
+	# put the new ones in
+	unless new_resource.filter.nil?
+		new_resource.filter.each do |filter|
+		if package_el.elements["filter[text() = '#{filter}']"].nil?
+			filter_el = package_el.add_element 'filter'
+			filter_el.add_text(filter)
+			end
+		end
+	end
+	unless new_resource.specifics.nil?
+		new_resource.specifics.each do |specific|
+			if package_el.elements["specific[text() = '#{specific}']"].nil?
+				specific_el = package_el.add_element 'specific'
+				specific_el.add_text(specific)
+			end
+		end
+	end
+	unless new_resource.include_ranges.nil?
+		new_resource.include_ranges.each do |r_begin, r_end|
+			if !package_el.nil? && package_el.elements["include-range[@begin = '#{r_begin}' and @end = '#{r_end}']"].nil?
+				package_el.add_element 'include-range', 'begin' => r_begin, 'end' => r_end
+			end
+		end
+	end
+	unless new_resource.exclude_ranges.nil?
+		new_resource.exclude_ranges.each do |r_begin, r_end|
+			if !package_el.nil? && package_el.elements["exclude-range[@begin = '#{r_begin}' and @end = '#{r_end}']"].nil?
+				package_el.add_element 'exclude_range', 'begin' => r_begin, 'end' => r_end
+			end
+		end
+	end
+	unless new_resource.include_urls.nil?
+		new_resource.include_urls.each do |include_url|
+			if package_el.elements["include-url[text() = '#{include_url}']"].nil?
+				include_url_el = package_el.add_element 'include-url'
+				include_url_el.add_text(include_url)
+			end
+		end
+	end
+	unless new_resource.store_by_if_alias.nil?
```

Comment:
`store_by_if_alias` is boolean, not an array, and has a default so you don't have to check for nil.